### DISCUSSION TITLE
Filesystem APi - filesystem.writeBinaryFile

### DISCUSSION THREAD

**Discussion by xarx** [neutralinojs/neutralinojs#670] [General]
Hello,

after i open a SQLite-File with: 

    let dbFile = await Neutralino.filesystem.readBinaryFile('./db.sqlite')

and i save the file without any mofification instant with: 

    let saveFile = await Neutralino.filesystem.writeBinaryFile('./newDB.sqlite', dbFile)

Neutralino saves the file, but this saved file is kind of "corrupted" (can't open this DB with any SQLite-Tools). At the Texteditor, both Files (original and saved) looks equal.

Whats going on here? :D

OS: Windows 11 / Standard-Browser: MS-Edge

**Comment by shalithasuranga** [comment]
Just tested on Linux by saving an image. looks like no issues. Sounds a platform-specific issue with Windows. Related with https://github.com/neutralinojs/neutralinojs/issues/619 seems

**Reply by xarx** [reply]
Hello Shalitha,

thank you very much for the quick reply. You are probably right, a Windows problem, I have since tried it on the Mac, it works there too (as it obviously does on Linux). On Windows the file gets 1 or 2 bytes bigger.

Do you have the time to take care of this? ;) Because especially the File-API makes your cool framework really interesting. And to a (if you _don't_ need the rest of Node) great alternative to the bloated Electron.

Otherwise at the same time many thanks for Neutralino, nice work! I discovered it only two days ago - but have been looking for such a solution for a long time. I like it a lot. Small, handy, focused on the essentials. Vue (which I work with most of the time) can be integrated easily - and all that without exhausting compiler intermediates. Really great.

**Reply by yapcheahshen** [reply]
did you binary compare the sqlite db file on different OS ? 
Agreed, a better FileAPI implmentation will make Neutralino a great platform to work on.
I would very much like to have "retain file handle" and "memory mapped file" as Neutralino has problem dealing with huge file now.
